rfc/text/wamp-terminology.md

```diff
@@ -0,0 +1,80 @@
+| *Broker*                          | *FIXME*                                                                                                 |
+| *Dealer*                          | *FIXME*                                                                                                 |
+| *Peer*                            | A WAMP *Client* or *Router*. An implementation might embed, provide or use both roles                   |
+| *Realm*                           | Isolated WAMP URI namespace, routing and administrative domain, optionally protected by **AA**          |
+| *Transport*                       | A message-based, reliable, ordered, bidirectional (full-duplex) channel over which *Peers* communicate  |
+| *Connection*                      | An underlying entity (if any) carrying the *Transport*, e.g. a network connection, pipe, queue or such  |
+| *Session*                         | Transient conversation between a *Client* and a *Router* on a *Realm* over a *Transport*                |
+| *Message*                         | *FIXME*                                                                                                 |
+| *Serializer*                      | *FIXME*                                                                                                 |
+| *Feature*                         | *FIXME*                                                                                                 |
+| *URI (Pattern)*                   | *FIXME*                                                                                                 |
+
+
+### Authentication and Authorization (AA)
+
+{align="left"}
+| Term                               | Definition                                                                                             |
+|------------------------------------|--------------------------------------------------------------------------------------------------------|
+| *Authentication*                   | Establishes the identity of a *Session* on a *Realm*                                                   |
+| *Principal*                        | Once authenticated, *Session*s identify under a *Principal*                                            |
+| *Credentials*                      | Any authentication information and secrets used during *Authentication*                                |
+| *Authorization*                    | Decides about permission for a given *Action* on an URI or URI pattern by a *Principal*                |
+| *Access Control*                   | Policy for selective restriction of access by *Action*s on URIs or URI patterns and *Principal*s       |
+| *Role-based Access Control (RBAC)* | An *Access Control* policy based on *Realm* (`realm`), *Role* (`authrole`) of *Principal* and *Action* |
+| *Discretionary Access Control*     | *FIXME*                                                                                                |
```

Comment:
The definition for this in the context of WAMP may be smth like this:

An *Access Control* policy based on lists or matrices of URIs and *Actions* mapped to *Principal*s

Reply:
How do you look at the "difference" between ACLs and AC-matrices?

I'm thinking like:

ACL: `(URI, Action) -> [Principal]`
ACM: `{Principal X Action} -> Actions`

Reply:
Yeah, I see ACL mostly the same. Also, ACLs may be expressed from the other side: `{Principal} -> [URI + Action]`

I see ACM more like this: `{Principal X URI} -> [Action]`

| | Uri 1 | Uri 2 |
| --- | --- | --- |
| Principal 1 | publish | call |
| Principal 2 | publish+subscribe | call+register |


But I think that all ways are eligible and as usual `it depends` :)